setuptools/config/setupcfg.py

```diff
@@ -280,7 +281,7 @@ def _section_options(
             yield name.lstrip('.'), value
 
     @property
-    def parsers(self):
+    def parsers(self) -> NoReturn:
```

Comment:
Ok, isn't this going to make more complicated inheritance and require some form of overload annotation later?

It is kind of an abstract method that needs to be implemented.
The expected signature is something like `() -> dict[str, Callable[[str], Any]]` or more specific[^1].
I suppose that for the sake of documentation/understanding that return value as a `dict` is better than `NoReturn`.

[^1]: The return values for the callable are probably something like all the types that can be produced by `ast.parse_literal` (or subtypes) + `packaging.specifiers.SpecifierSet`

Reply:
`NoReturn` is a bottom type, so as far as subclassing it shouldn't be an issue.

For consumers it means that if something returns a `ConfigHandler`, they'd see accessing the property `parsers` as never returning (unless they use `isinstance` to ensure a subtype or something).

So you are correct that 
> I suppose that for the sake of documentation/understanding that return value as a `dict` is better than `NoReturn`.

Even better, this property should be marked abstract, so type-checkers tells the user to never instanciate `ConfigHandler`, which implies that any `ConfigHandler` the user gets should be a concrete subclass, so returning `NoReturn` for a `NotImplementedError` is no longer relevant.

But that's for a separate PR ! (I'll just revert this line here)